File: JINSEOP/프로그래머스_WEEK14/시저 암호.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

void encrypt(string& s, int n) {
    for (char& c : s) {
        if ('A' <= c && c <= 'Z')
            c = (c + (n - 'A')) % 26 + 'A';
        if ('a' <= c && c <= 'z')
            c = (c + (n - 'a')) % 26 + 'a';
    }
}

string solution(string s, int n) {
    string answer = "";

    encrypt(s, n);
    answer = s;

    return answer;
}

File: DONGUK/프로그래머스_WEEK_14/시저 암호.py
def solution(s, n):
    answer = ''
    
    for i in range(len(s)):
        if s[i] == ' ':
            answer += s[i]
            continue
            
        else:
            n_s = ord(s[i]) + n
            
            if 'A' <= s[i] <= 'Z':
                if n_s > 90:
                    n_s -= 26
            
            if 'a' <= s[i] <= 'z':
                if n_s > 122:
                    n_s -= 26
                
            answer += chr(n_s)
                  
    return answer